src/test/HostActivationTests/FrameworkResolution/FrameworkResolutionBase.cs

```diff
@@ -24,23 +24,85 @@ public abstract class FrameworkResolutionBase
             string[] commandLine = null,
             bool multiLevelLookup = false)
         {
-            runtimeConfig(RuntimeConfig.Path(app.RuntimeConfigJson)).Save();
+            RunTest(
+                dotnet,
+                app,
+                new TestSettings()
+                {
+                    RuntimeConfigCustomizer = runtimeConfig,
+                    Environment = environment,
+                    CommandLine = commandLine
+                },
+                resultAction,
+                multiLevelLookup);
+        }
+
+        public class TestSettings
+        {
+            public Func<RuntimeConfig, RuntimeConfig> RuntimeConfigCustomizer { get; set; }
+            public IEnumerable<string> Environment { get; set; }
```

Comment:
Could we just store this an IDictionary instead of adding the = and then splitting by the = later?